images/airflow/2.9.2/python/mwaa/entrypoint.py

```diff
@@ -116,6 +117,37 @@ async def airflow_db_init(environ: dict[str, str]):
     await run_command("python3 -m mwaa.database.migrate", env=environ)
 
 
+async def increase_pool_size_if_default_size(environ: dict[str, str]):
+    """
+    Update the default pool size
+
+    Fix a regression where some 2.9.2 environments were created with the default 128 default
+    pool size. This function checks if the environment was created during the problematic
+    timeframe and update the size if it has not been updated by the customer.
+
+    :param environ: A dictionary containing the environment variables.
+    """
+    created_at = os.environ.get("MWAA__CORE__CREATED_AT")
+
+    if created_at:
+        date_format = "%a %b %d %H:%M:%S %Z %Y"
+        created_date = datetime.strptime(created_at, date_format)
+        # Has a little buffer from when 2.9.2 was released and when the fix was fully deployed
+        issue_beginning = datetime(2024, 7, 8)
+        issue_resolution = datetime(2024, 9, 6)
+
+        if created_date > issue_beginning and created_date < issue_resolution:
+            command_output = []
+
+            # Get the current default_pool size
+            await run_command("airflow pools get default_pool | grep default_pool | awk '{print $3}'",
```

Comment:
Is there a reason we can't use Pool.get_pool("default_pool") instead? Reference: https://github.com/apache/airflow/blob/2.9.2/airflow/models/pool.py#L66.

I don't doubt this works, but it's somewhat opaque as-is.

Reply:
I remember trying to use this but I think there was some issues that came up with setting up the session since we are not running it as a dag.

Reply:
I see, makes sense. If you've time boxed it and could not get it to work with Pools, I think it's okay to leave it like this.